Document scoped API keys (#9)
whopio/api-docs: src/pages/api-keys.mdx

import { Warning } from '../components/mdx'

# API keys

> API keys are an essential tool for accessing and integrating with Whop's API. An API key is a unique identifier that allows a software product to access Whop's API and validate licenses, retrieve user information, and more.

Reviewer: ```suggestion
API keys are an essential tool for accessing and integrating with Whop's API. An API key is a unique identifier that allows a developer to access Whop's API on behalf of a company and validate licenses, retrieve user information, and more.
```

<Warning>

>   It's important that you keep your API keys safe and secure - unless otherwise

Reviewer: It's important that you keep your API keys safe and secure - if not properly scoped, your API keys can be dangerous if they fall into the wrong hands.

  set, your API keys carry a lot of permissions that could be potentially
  destructive if gotten into the wrong hands!
</Warning>

## Creating an API key

1. Head to your Developers setting panel under your business
2. Click "New API Key"
3. Set a memorable name for your key.
4. Copy the API key

After that, you're ready to start creating with Whop's API!

## Scoped API keys

Scoped API keys provide access to specific resources or endpoints within Whop's API, rather than providing access to the entire API. This helps to ensure that the client-side code is only able to perform the specific actions that it needs to, while also limiting the potential for security vulnerabilities.

> Scoped API keys are **the only way** for your company to use client-sided keys.

Reviewer: ```suggestion
Scoped API keys are **the best way** for your company to use client-sided keys.
```

![Scoped Key Setup](https://i.imgur.com/sbMzXjv.gif){{ className: 'rounded-lg w-full' }}

### Instructions to create a scoped API key

1. Create a new API key under the Developers setting panel
2. Click the triple dots and click "Edit permissions"

> 3. Add the permissions you want this key to have access to

Reviewer: ```suggestion
3. Add the permissions you want this key to have access to. If you are storing keys client-side, we recommend keeping your permissions as minimal as possible. 
```

4. Save!

## Further help

> For more authentication help, [our developer documentation page on authentication](https://dev.whop.com/reference/authentication) can provide some more technical specifics.

Reviewer: ```suggestion
For more authentication help, [our API reference page on authentication](https://dev.whop.com/reference/authentication) can provide some more technical specifics.
```
